src/rules/noNamespaceRule.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright 2016 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../lint";
+import {isNestedDeclaration} from "./noInternalModuleRule";
+
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "'namespace' and 'module' are disallowed";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new NoNamespaceWalker(sourceFile, this.getOptions()));
+    }
+}
+
+class NoNamespaceWalker extends Lint.RuleWalker {
+    public visitModuleDeclaration(decl: ts.ModuleDeclaration) {
+        super.visitModuleDeclaration(decl);
+        // declare module 'foo' {} is an external module, not a namespace.
+        if (decl.name.kind === ts.SyntaxKind.StringLiteral) { return; }
+        if (Lint.isNodeFlagSet(decl, ts.NodeFlags.Ambient) && this.hasOption("allow-declarations")) { return; }
+        if (isNestedDeclaration(decl)) { return; }
```

Comment:
And this can become `Lint.isNestedModuleDeclaration` with the change mentioned above